### DISCUSSION TITLE
Brown out on excessiv Web UI usage

### DISCUSSION THREAD

**Discussion by spanzetta** [jomjol/AI-on-the-edge-device#2854] [General]
### The Problem

After the update to latest 15.5 I am experiencing totally hangs (it doesn't even respond to ping) after few minutes of running..
I cannot find anything relevant in the logs (standard logs.. )
Anyone else with same problem?

### Version

15.5

### Logfile

```shell
no log
```

**Comment by spanzetta** [comment]
After reflashed the ESP (with webtools) the problem is solved..
Maybe something did went wrong with the update

**Comment by friedpa** [comment]
15.5.0 Working without any problems so far

**Comment by spanzetta** [comment]
I spoke too early.. it hangs after 2/3 rounds
I see a red led flashing 3 times.. pause..  then 2 times.. and then again
Need to understand what does it mean

**Comment by Slider0007** [comment]
Check the docu: https://jomjol.github.io/AI-on-the-edge-device-docs/StatusLED-BlinkCodes/

Seems a SD card issue...

**Comment by spanzetta** [comment]
This is what happens now..
Does it makes sense?
With previous 15.4 it was working almost perfectly (running from few days)

![image](https://github.com/jomjol/AI-on-the-edge-device/assets/49818743/ce252e31-3e46-4bf8-8596-2e91e1daa4a6)

**Comment by spanzetta** [comment]
I deleted the current data log txt file (it was containing some data with 1-1-1970 date)... and now it appears the value.,. but this is also with no sense

![image](https://github.com/jomjol/AI-on-the-edge-device/assets/49818743/56cc1070-6ebd-4128-8aef-d3e4668cd1be)

**Comment by spanzetta** [comment]
> Check the docu: https://jomjol.github.io/AI-on-the-edge-device-docs/StatusLED-BlinkCodes/
> 
> Seems a SD card issue...

Yes.. I found that info too .. (thanks)
But it is very strange.. the problem of hanging happens ONLY when accessing to the Data -> Recognition tab..
How is possible that only in that case the SD card has problems?

**Comment by spanzetta** [comment]
Why the raw value is correct.. but the value is not shown?

![image](https://github.com/jomjol/AI-on-the-edge-device/assets/49818743/bbaaac25-5769-4cc4-81d5-b2b4da62161d)

**Comment by friedpa** [comment]
Read the Value Status! Rate too high...
Set the Prevalue new under the Tab Settings!

**Reply by spanzetta** [reply]
I do.. but at next read cycle again the value is wrong (one digit more) even if raw value is correct!
That's something that wasn't happening before with 15.4

**Reply by spanzetta** [reply]
Again.. why the raw value is correct.. but value is wrong? 
Does it mean that postprocessing is increasing value by one?
How to avoid that?

**Reply by spanzetta** [reply]
After have changed again and again the ROI digit position/size.. it is now detecting properly: it is so sensitive that even changing one or two pixels can make the difference in proper/not proper detection

**Reply by spanzetta** [reply]
Continuing to learn how to use this fantastic device ;-)

**Comment by nliaudat** [comment]
> Seems a SD card issue...

Perhaps related to ATA-Trim support addon in https://github.com/jomjol/AI-on-the-edge-device/commit/2ed6fb0f0dc77246265ac8c8019414fe0e197654 https://github.com/jomjol/AI-on-the-edge-device/pull/2781

**Reply by spanzetta** [reply]
Hmm.. so it can be that now I start to have problems with my sdcard because of this change?

**Comment by caco3** [comment]
Also make sure you have a good power supply and a good SD-Card!

**Reply by spanzetta** [reply]
Sure.. they are both good and I never had problems until now (until they are now not good anymore as coincidence)

**Comment by spanzetta** [comment]
What does it mean ?
_[0d00h00m05s] 2024-02-03T09:18:54 <INF> [MAIN] Reset reason: Brownout_
After 5 seconds?

**Reply by spanzetta** [reply]
Found..
1.0.1.1 Overload during HTML access

**Reply by spanzetta** [reply]
I've now learned that don't have to play too fast with web interface;-)

**Reply by nliaudat** [reply]
>Reset reason: Brownout

In general, that's linked to an insufficient power supply

**Reply by spanzetta** [reply]
I don't think it's my case..
If I am "delicate" using web interface it's stable

I am using 220v converted to 5v (5W) with this product 
https://it.aliexpress.com/item/1005006100095956.html?spm=a2g0n.order_detail.order_detail_item.3.6f4e582fHTuPSc&gatewayAdapt=glo2ita

**Reply by caco3** [reply]
a brown out always means your power supply is not good enough!

**Reply by nliaudat** [reply]
HLK modules do not have capacitor to smooth the power consumption. 
It's recommended to add one near the board.

**Reply by spanzetta** [reply]
> HLK modules do not have capacitor to smooth the power consumption.
> It's recommended to add one near the board.

hmm.. didn't know that .. can you pls explain it better? 
what kind of capacitor and how to connect?

Consider that from HLK to the ESP32 there is a 7/8 meters cable.. (my water meter is in the garden)

**Reply by nliaudat** [reply]
Aluminium electrolytic capacitance,
capacity 100-220 UF, voltage
reduction greater than 75%

have a look at https://datasheet.lcsc.com/lcsc/2310091637_HI-LINK-HLK-PM03_C209904.pdf

DC current is very sensitive to the type of cable used to carry it. The lower the resistance of the cable, the lower the loss. The type of resistor depends on whether you are using 5 V or 3.3 V. I recommend using 230V in your garden, then converting to 5v with a simple USB transformer.

**Reply by caco3** [reply]
Also, I would not trust any 230V device from Ali Express! I have seen stuff from there I would not even power in my lab just for testing.

Like @nliaudat sayd, you could pull 230V close to your device if possible. Alternatively (how I do such things), have a 12V power supply which provides the power close to the device. Due to the length and cable lose, it might only be 9..10V at the end. But that is not a problem. use a power regulator like a LM7805 to convert it to the 5V and use some block condensators to stabilize the power.

**Reply by spanzetta** [reply]
Many thanks for all these suggestion and precious information..
In the meantime I can tell you that, without using the web interface (I push data with MQTT to my HA) the ESP32 is running from almost one day without rebooting..

See below
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/49818743/76cb6521-7995-4868-8fc4-5d013a570cd9)

**Comment by spanzetta** [comment]
Update: 

as test I tried to access the web interface after about 20 hours of uninterrupted running.. and it immediately restarted..

Considering that the same setup/same powering was running without problems for over a week with 15.4 version (i was accessing web interface very intensively without any issue).. and now with 15.5 I have this issue I can assume/conclude that this version is much more power demanding..

Does it makes sense?

**Reply by caco3** [reply]
Do watever you want, but it is clear that your power supply setup is indufficient.

**Reply by spanzetta** [reply]
Now it is..
Before (15.4) it wasn't

**Reply by spanzetta** [reply]
On the other side he good thing of 15.5 is that there is no more memory leak..
Free memory is stable across time
